Gds/src/fprime_gds/__init__.py

```diff
@@ -1,2 +1,7 @@
 # Python 2/3 compatibility for namespace packages
 __path__ = __import__("pkgutil").extend_path(__path__, __name__)
+
+# Currently the GDS is backwards compatible with all dictionaries. This MUST be bumped with each framework release to
+# ensure compatibility
+MINIMUM_SUPPORTED_FRAMEWORK_VERSION = "0.0.0"
+MAXIMUM_SUPPORTED_FRAMEWORK_VERSION = "1.5.4"
```

Comment:
There are different ways to do this:

**Method 1:**
One way to define package version is to define it as `__version__`. See [package-version](https://packaging.python.org/guides/single-sourcing-package-version/)

Define `version.py` in `Gds/src/fprime_gds` with the following content:
```python
__min_version__ = (0, 0, 0)
__max_version__ = (1, 5, 4)
__version__ = '.'.join(map(str, __max_version__))
```
Then import it in Gds/src/fprime_gds/__init__.py as
```python
from .version import __min_version__
from .version import __max_version__
from .version import __version__
```
Now one can get the version of the package by calling 
```python
import fprime_gds
fprime_gds.__version__
```
output would be `'1.5.4'`

to get the version in Gds/src/fprime_gds/common/loaders/xml_loader.py to check the major release we can write
```python
import fprime_gds
dict_ver = tuple(map(int, root.attrib.get("framework_version", "1.2.3").split('.')))[0:2]
min_ver = fprime_gds.__min_version__[0:2]
max_ver = fprime_gds.__max_version__[0:2]

if any([dict_ver < min_ver, dict_ver >  max_ver]):
...
```
Also in the setup.py we can have:
```
exec(open('src/fprime_gds/version.py').read())

setup(
version=__version__,
)
```
This way the setup.py version will be updated when the version.py is updated.

**Method 2:**
Another option is to use importlib.metadata
this library will look at version attribute in setup.py

To get the version in Gds/src/fprime_gds/common/loaders/xml_loader.py one can write:
```
from importlib import metadata
import fprime_gds
if metadata.version('fprime_gds') > "2.2":
...
```

For more resource on defining metadata see [core-metadata](https://packaging.python.org/specifications/core-metadata/)

Reply:
I like your suggestion of making these versions tuples.  Easier to handle.  I will do that.